src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java

```diff
@@ -338,14 +339,13 @@ void maybeAddPartitions(int minPartitionNum) throws ExecutionException, Interrup
       int partitionDifference = minPartitionNum - partitionNum;
 
       // leader assignments -
-      for (BrokerMetadata brokerMetadata : brokers) {
+      while (newPartitionAssignments.size() != partitionDifference) {
         List replicas = new ArrayList<>();
         // leader replica/broker -
-        replicas.add(brokerMetadata.id());
+        int brokerMetadata = randomBroker(brokers).id();
+        replicas.add(brokerMetadata);
```

Comment:
Since randomBroker will randomly return one, it is possible that two same brokers are returned right?

Reply:
yes, that won't be a problem

Reply:
as the leadership transition and partition reassignments occur on an interval periodically.

Reply:
@yifangl has a good point. The replicas array must contain distinct broker ids. random does not guarantee that.